src/smc-webapp/project_files.cjsx

```diff
@@ -1525,44 +1525,66 @@ ProjectFilesActionBox = rclass
             if not (single_file_data.is_public and single_file_data.public?.path isnt single_file)
                 @share_click()
 
-    download_click: ->
+    download_single_click: ->
         @props.actions.download_file
             path : @props.checked_files.first()
             log : true
         @props.actions.set_file_action()
 
-    render_download_link: (single_item) ->
+    download_multiple_click: ->
+        destination = ReactDOM.findDOMNode(@refs.download_archive).value
+        @props.actions.zip_files
+            src  : @props.checked_files.toArray()
+            dest : misc.path_to_file(@props.current_path, destination)
+        @props.actions.download_file
```

Comment:
> What do you mean by this?

look at `zip_files` in `webapp_client.exec`. not only is there no callback, but the one used in `webapp_client.exec` is set to `@_finish_exec(id)`. So, to me this looks like `@_finish_exec(id)` needs a second argument to be able to trigger an additional callback after the finish-exec callback. :arrow_right:  :stuck_out_tongue_winking_eye: 

(If this makes no sense to you, don't worry -- and there are maybe better ideas, too...)